### PR TITLE
Fix `get_custody_columns`

### PR DESCRIPTION
**Pull request by hwwhww**
Replace #3742
Thank @nalepae for raising the issue, and thank @zilm13 for proposing a fix for the bug. 🙏

### CODE REVIEW

File: specs/_features/eip7594/das-core.md [hunk 1]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
     assert custody_subnet_count <= DATA_COLUMN_SIDECAR_SUBNET_COUNT
 
     subnet_ids: List[uint64] = []
-    i = 0
+    tmp_id = uint256(node_id)
```
Comment (ralexstokes) [inline specs/_features/eip7594/das-core.md:109]:
```suggestion
    i = uint256(node_id)
```

File: specs/_features/eip7594/das-core.md [hunk 2]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
...
     while len(subnet_ids) < custody_subnet_count:
-        if node_id == UINT256_MAX:
-            node_id = NodeID(0)
+        # Overflow prevention
+        if tmp_id == UINT256_MAX:
+            tmp_id = NodeID(0)
```
Comment (ralexstokes) [inline specs/_features/eip7594/das-core.md:113]:
```suggestion
        if i == UINT256_MAX:
            i = NodeID(0)
```

File: specs/_features/eip7594/das-core.md [hunk 3]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
...
 
         subnet_id = (
-            bytes_to_uint64(hash(uint_to_bytes(uint256(node_id + i)))[0:8])
+            bytes_to_uint64(hash(uint_to_bytes(uint256(tmp_id)))[0:8])
```
Comment (ralexstokes) [inline specs/_features/eip7594/das-core.md:116]:
```suggestion
            bytes_to_uint64(hash(uint_to_bytes(uint256(i)))[0:8])
```

File: specs/_features/eip7594/das-core.md [hunk 4]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
...
             % DATA_COLUMN_SIDECAR_SUBNET_COUNT
         )
         if subnet_id not in subnet_ids:
             subnet_ids.append(subnet_id)
-        i += 1
+        tmp_id += 1
```
Comment (ralexstokes) [inline specs/_features/eip7594/das-core.md:121]:
```suggestion
        i += 1
```

**Review by ralexstokes — APPROVED** [review]
I would maybe just leave this variable as `i`, otherwise it looks good!

also could consider mixing in a `%` operator instead, but it would end up having the same effect as this code, and this seems fine enough

File: specs/_features/eip7594/das-core.md [hunk 5]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
     assert custody_subnet_count <= DATA_COLUMN_SIDECAR_SUBNET_COUNT
 
     subnet_ids: List[uint64] = []
-    i = 0
+    tmp_id = uint256(node_id)
     while len(subnet_ids) < custody_subnet_count:
-        if node_id == UINT256_MAX:
-            node_id = NodeID(0)
+        # Overflow prevention
+        if tmp_id == UINT256_MAX:
```
Comment (zilm13) [inline specs/_features/eip7594/das-core.md:112]: It makes more sense to move this part of the code below, just before the increment like @dankrad did in #3752, so we will not skip uint256_max iteration step

File: specs/_features/eip7594/das-core.md [hunk 6]
```diff
@@ -106,18 +106,20 @@ def get_custody_columns(node_id: NodeID, custody_subnet_count: uint64) -> Sequen
     assert custody_subnet_count <= DATA_COLUMN_SIDECAR_SUBNET_COUNT
 
     subnet_ids: List[uint64] = []
-    i = 0
+    i = uint256(node_id)
```
Comment (hwwhww) [inline specs/_features/eip7594/das-core.md:109]: Ah sorry @ralexstokes I think I have the revert the previous commit. There is another `i` usage in the `return sorted(...)`... so it's less good to assign an `i` here.

**Pull request merged**